### PR TITLE
initial commit of related features sample

### PR DESCRIPTION
**Pull request by ldanzinger**
@nandinirao please review/merge

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/Features/ListRelatedFeatures/RelatedFeature.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#define RELATEDFEATURE_H
+
+#include <QString>
+
+class RelatedFeature
+{
+
+public:
+  explicit RelatedFeature(const QString& displayFieldName, const QString& displayFieldValue,
+                      const QString& serviceLayerName);
```
Comment (nandinirao) [inline ArcGISRuntimeSDKQt_CppSamples/Features/ListRelatedFeatures/RelatedFeature.h]: Alignment is not right here

File: ArcGISRuntimeSDKQt_CppSamples/Features/ListRelatedFeatures/ListRelatedFeatures.cpp [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                  m_relatedFeaturesModel->addRelatedFeature(relatedFeature);
+                  emit relatedFeaturesModelChanged();
+                }
+              }
+
+              emit showAttributeTable();
+              m_mapView->setViewpointGeometry(arcGISFeature->geometry(), 100);
+            });
+
+            // query related FeatureServiceCapabilities
```
Comment (nandinirao) [inline ArcGISRuntimeSDKQt_CppSamples/Features/ListRelatedFeatures/ListRelatedFeatures.cpp]: Comment should be fixed here.

File: ArcGISRuntimeSDKQt_QMLSamples/Features/ListRelatedFeatures/ListRelatedFeatures.qml [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+                                };
+                                relatedFeaturesModel.append(listElement);
+                            }
+                        }
+
+                        // show the attribute view
+                        attributeView.height = 200 * scaleFactor
+                    });
+
+                    // query related FeatureServiceCapabilities
```
Comment (nandinirao) [inline ArcGISRuntimeSDKQt_QMLSamples/Features/ListRelatedFeatures/ListRelatedFeatures.qml]: Same here, may be just say query related feautures?

**Comment by ldanzinger** [comment]
Thanks @nandinirao. Should be updated now

**Pull request merged**